py/objstr.c

```diff
@@ -832,6 +832,10 @@ STATIC bool arg_looks_numeric(mp_obj_t arg) {
     ;
 }
 
+STATIC bool arg_looks_string(mp_obj_t arg) {
+    return MP_OBJ_IS_TYPE(arg, &mp_type_str) || MP_OBJ_IS_STR(arg);
```

Comment:
@pfalcon 
My fault, I've just copy logic from similar functions. So I change this to `MP_OBJ_IS_TYPE(arg, &mp_type_str)`
If you think it's better to make check explicitly (without extra function) I will change pull.
Also, if you think, that name of commit is bad, please tell what is ok for this.

Logic is simple. If argument is not number, and format string does not have `type` field, and argument is `string`, then `type` should be set as string. It will be `plain object` else case and we will have no formatting at all, as we can see in #1708